app/src/userpages/modules/userPageStreams/reducer.js

```diff
@@ -271,12 +269,13 @@ export default function (state: UserPageStreamsState = initialState, action: Str
             }
 
         case UPDATE_EDIT_STREAM_FIELD: {
-            const newState = {
+            return {
                 ...state,
+                editedStream: {
+                    ...state.editedStream,
+                    [action.field]: action.data,
+                },
             }
-            const fullPath = `editedStream.${action.field}`
-            set(newState, fullPath, action.data)
-            return newState
```

Comment:
Here! `newState.editedStream` was ALWAYS referencing the same object in memory. This is the reason why `useSelector` ignored updates to both name and description.

`useSelector` takes a second argument – `equalityFn`, `(a, b) => a === b` by default. `A === A` is always true. And using [`shallowEqual`](https://github.com/reduxjs/react-redux/blob/master/src/utils/shallowEqual.js) won't work either for obvious reasons (it's `A === A` but proves that one can write a loop).

The only answer is to.. update the store properly.

A cute little buggy bug. 🐞

Reply:
Ahh interesting! Nice one.. I need to learn up on how this useSelector works with state updates..

Although seems to affect the configure fields section below which uses the same `UPDATE_EDIT_STREAM_FIELD` - if you 'Add field' it doesn't update the config fields state properly:

Dev branch:
<img width="1180" alt="Screenshot 2019-12-17 at 20 51 50" src="https://user-images.githubusercontent.com/1593398/71025962-d8e06780-2110-11ea-9d05-0400750fb245.png">

This branch:
<img width="1172" alt="Screenshot 2019-12-17 at 20 49 25" src="https://user-images.githubusercontent.com/1593398/71025980-e138a280-2110-11ea-8ec7-2601095f546c.png">

The `development` branch has it's own bugs of course I noticed as well after deploying to production :P Seems as though the hooks refactoring has had some collateral damage!
https://streamr.atlassian.net/browse/PLATFORM-1183